### PR TITLE
Kneaded the repository 🍞 and  added to `Makefile`

### PR DESCRIPTION
**Pull request by michplunkett**
## Describe your changes
I checked to see if they had formatting for Jupyter notebooks and it ends up they do! `jupyter-black` syncs up with the `[tool.black]` section in the `pyproject.toml` file for configuration things.

If you'd like to see this work in a new Jupyter notebook, you just need this code at the import spot of the file:
```python
import jupyter_black
.
.
.
jupyter_black.load()
```

## Checklist before requesting a review
- [x] The code runs successfully.

<details><summary>make run-all-models and run prepare-data commands</summary>

```
(poetry-python-project-py3.11) michaelp@MacBook-Air-3 supreme-court-ml-predictions % make run-all-models
python -m "supreme_court_predictions" --logistic-regression --random-forest --xg-boost
oh, what up?
Creating bag of words
Starting the Logistic Regression
------------------------------------------
Running a regression on case_aggregations...
Accuracy score: 0.5405405405405406
------------------------------------------
Creating bag of words
Starting the Logistic Regression
------------------------------------------
Running a regression on judge_aggregations...
Accuracy score: 0.5205479452054794
------------------------------------------
Creating bag of words
Starting the Logistic Regression
------------------------------------------
Running a regression on advocate_aggregations...
Accuracy score: 0.7567567567567568
------------------------------------------
Creating bag of words
Starting the Logistic Regression
------------------------------------------
Running a regression on adversary_aggregations...
Accuracy score: 0.7837837837837838
------------------------------------------
Creating bag of words
Starting the Random Forest
------------------------------------------
Running a random forest on case_aggregations...
Accuracy score: 0.6216216216216216
------------------------------------------
Creating bag of words
Starting the Random Forest
------------------------------------------
Running a random forest on judge_aggregations...
Accuracy score: 0.6438356164383562
------------------------------------------
Creating bag of words
Starting the Random Forest
------------------------------------------
Running a random forest on advocate_aggregations...
Accuracy score: 0.8108108108108109
------------------------------------------
Creating bag of words
Starting the Random Forest
------------------------------------------
Running a random forest on adversary_aggregations...
Accuracy score: 0.8378378378378378
------------------------------------------
Creating bag of words
Starting the XGBoost model
------------------------------------------
Error: training data is not big enough for this subset
------------------------------------------
Creating bag of words
Starting the XGBoost model
------------------------------------------
Error: training data is not big enough for this subset
------------------------------------------
Creating bag of words
Starting the XGBoost model
------------------------------------------
Error: training data is not big enough for this subset
------------------------------------------
Creating bag of words
Starting the XGBoost model
------------------------------------------
Error: training data is not big enough for this subset
------------------------------------------
(poetry-python-project-py3.11) michaelp@MacBook-Air-3 supreme-court-ml-predictions % make prepare-data
python -m "supreme_court_predictions" --clean-data --describe-data --tokenize-data --process-data
oh, what up?
Working in /Users/michaelp/Documents/GitHub/supreme-court-ml-predictions/supreme_court_predictions/data/convokit/
Data will be saved to /Users/michaelp/Documents/GitHub/supreme-court-ml-predictions/supreme_court_predictions/data/clean_convokit/
Parsing cases...
Parsing speakers...
Parsing conversations metadata...
Parsing utterances...
Data saved to /Users/michaelp/Documents/GitHub/supreme-court-ml-predictions/supreme_court_predictions/data/clean_convokit/
Data will be saved to: 
/Users/michaelp/Documents/GitHub/supreme-court-ml-predictions/supreme_court_predictions/data/clean_convokit/
Spacy module successfully loaded.
Spacy tokenization complete.
Working in /Users/michaelp/Documents/GitHub/supreme-court-ml-predictions/supreme_court_predictions/data/clean_convokit/
Data will be saved to /Users/michaelp/Documents/GitHub/supreme-court-ml-predictions/supreme_court_predictions/data/statistics/
Grabbing case descriptive statistics...
Grabbing advocates descriptive statistics...
Grabbing speakers descriptive statistics...
Grabbing voters descriptive statistics...
Grabbing utterances descriptive statistics...
Data saved to /Users/michaelp/Documents/GitHub/supreme-court-ml-predictions/supreme_court_predictions/data/statistics/
Working in /Users/michaelp/Documents/GitHub/supreme-court-ml-predictions/supreme_court_predictions/data/clean_convokit/
Data will be saved to /Users/michaelp/Documents/GitHub/supreme-court-ml-predictions/supreme_court_predictions/data/processed/
Grabbing token aggregation for all cases...
Grabbing token aggregation for advocates...
Grabbing token aggregation for adversaries...
Grabbing token aggregation for judges...
Exporting files...
Data saved to /Users/michaelp/Documents/GitHub/supreme-court-ml-predictions/supreme_court_predictions/data/processed/
(poetry-python-project-py3.11) michaelp@MacBook-Air-3 supreme-court-ml-predictions % 
```
</details>

- [x] I ran `make lint` and have made the bulk of changes that it has requested.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -21,12 +21,14 @@ spacy = "^3.5.2"
 ruff = "^0.0.263"
 xgboost = "^1.7.5"
 jupyter = "^1.0.0"
+jupyter-black = "^0.3.4"
```
Comment (michplunkett) [inline pyproject.toml:24]: Add the package.

File: Makefile [hunk 2]
```diff
@@ -44,10 +40,26 @@ tokenize-data:
...
 random-forest:
 	python -m ${BASEDIR} --random-forest
+
+.PHONY: xg-boost
+random-forest:
+	python -m ${BASEDIR} --xg-boost
+
+.PHONY: run
+run:
+	make clean-data prepare-data run-all-models
```
Comment (chanteriam) [inline Makefile:61]: makes sense!

File: supreme_court_predictions/models/model.py [hunk 3]
```diff
@@ -44,18 +44,20 @@ def __repr__(self):
         Overwrites default string representation
         """
 
-    @staticmethod
-    def print_results(model_name="", accuracy_score=None, dataframe_name=None):
```
Comment (michplunkett) [inline supreme_court_predictions/models/model.py:48]: I needed the `self` value so I took away the `@staticmethod` tag.

File: Makefile (hunk 2 above)
Comment (michplunkett) [reply Makefile]: This way we can keep the commands for each function, but also keep the `run` command is somewhere under 40k characters long.

image

**Review by FedericoDM — APPROVED** [review]
:)

**Review by JJModern — APPROVED** [review]
looks good!

**Pull request merged**